examples/js/loaders/GLTFLoader.js

```diff
@@ -1454,6 +1539,111 @@ THREE.GLTFLoader = ( function () {
 
 	};
 
+	/**
+	 * @param {string} key
+	 * @param {number} index
+	 * @param {GLTF.definition} def
+	 */
+	GLTFParser.prototype._onBefore = function ( key, index, def ) {
+
+		var functionName = 'onBefore' + key[ 0 ].toUpperCase() + key.slice( 1 );
+		var plugins = this.plugins || {};
+
+		for ( var pluginName in plugins ) {
+
+			var plugin = plugins[ pluginName ];
+
+			if ( plugin[ functionName ] === undefined ) continue;
+
+			plugin[ functionName ]( index, def, this );
+
+		}
+
+	};
+
+	/**
+	 * @param {string} key
+	 * @param {Object} object
+	 * @param {number} index
+	 * @param {GLTF.definition} def
+	 * @return {Promise<Object>}
+	 */
+	GLTFParser.prototype._onAfter = function ( key, object, index, def ) {
+
+		var functionName = 'onAfter' + key[ 0 ].toUpperCase() + key.slice( 1 );
+		var plugins = this.plugins || {};
+		var targetPlugins = [];
+
+		for ( var pluginName in plugins ) {
+
+			var plugin = plugins[ pluginName ];
+
+			if ( plugin[ functionName ] !== undefined ) {
+
+				targetPlugins.push( plugin );
+
+			}
+
+		}
+
+		return _onAfterHelper( targetPlugins, functionName, object, index, def, this );
+
+	};
+
+	function _onAfterHelper( plugins, functionName, object, index, def, parser ) {
+
+		while ( plugins.length > 0 ) {
+
+			var plugin = plugins.shift();
+			var result = plugin[ functionName ]( object, index, def, parser );
+
+			if ( result instanceof Promise ) {
+
+				object = result;
+
+			} else {
+
+				return result.then( function ( object ) {
+
+					return _onAfterHelper( plugins, object, functionName, index, def, parser );
+
+				} );
+
+			}
+
+		}
+
+		return Promise.resolve( object );
+
+	}
+
```

Comment:
`parser._onAfter()` and `_onAfterHelper()` may look a bit tricky. The background is I want to run `onAfterXXX` of plugins sequentially because in the current API proposal we allow plugin's `onAfterXXX` to asynchronously create a new object. (If we could use await, I could write much simpler?)

Another solution to simplify the code may be to force plugin's `onAfterXXX` to always return `Promise<Object>`. But now `onAfterXXX` is fired regardless of extension definition so if we do that parser can unnecessarily form a long promise chain which can have a big response time impact. I wanted to avoid that.

Any feedback to simplify is very welcome.